examples/incorrect/function-overloading-visibility-1.ccp

```diff
@@ -0,0 +1,8 @@
+# shadowing variable does not retrieve old overloads
+let f = [] -> Bool => true;
+let f : Int = 7;

```

Comment:
My impression was that we put functions and variables int the same namespace, so that we could have first-class functions.

So the error would actually be already at this point, when we have a variable and a function with the same name?